frontend/src/components/NotesDialog.js

```diff
@@ -7,16 +7,21 @@ import DialogContent from "@material-ui/core/DialogContent"
 import DialogContentText from "@material-ui/core/DialogContentText"
 import DialogTitle from "@material-ui/core/DialogTitle"
 import TextField from "@material-ui/core/TextField"
+import { QueueStoreContext } from "../stores/QueueStore"
 
-function NotesDialog({ visibleDialog, toggleVisibleDialog }) {
+function NotesDialog({ visibleDialog, toggleVisibleDialog, queueData, id }) {
+  const queueStore = useContext(QueueStoreContext)
   const [participantNotes, setParticipantNotes] = React.useState("")
 
-  const handleSubmit = () => {
-    // Make POST request to API here to submit data
-    // Need to pass in visit ID to this component (most likely) for API
+  const handleSubmit = (queueId, visitId) => {
+    queueStore.patchVisit(queueId, visitId, { notes: participantNotes })
     toggleVisibleDialog()
   }
 
+  useEffect(() => {
+    setParticipantNotes("")
```

Comment:
Try the following
```
const { getNotes } = queueStore;
...
useEffect(() => {
  setParticipantNotes(getNotes(queueData, id));
}, [getNotes, queueData, id]);

```

I also don't see an `onChange` handler or value set for the `TextField` to use `participantNotes`

Reply:
Excellent.  Easy once you've explained it! I got the mobx error I described above if I used the assignment for getNotes, but works if I use queueStore.getNotes

Reply:
I got rid of the field above the input showing the previous comment, since it's now visible already. I don't think an onChange is needed. Do you?

Reply:
How do you update participantNotes without an onChange? Some lines of code may be missing in this review because I don't even see the TextField's `/>` .